Fix `ModelDebugger.step` to use only `intermediate_outputs`

Zerui18 opened the pull request:
### Issue
Calling `cto.coreml.experimental.linear_quantize_activations` on a model with `ct.ImageType` output(s) fails at the model output because `record_intermediate_output` expects `output_value` to be a `np.ndarray` but the model's final output is `PIL.Image`.

Ultimately, `record_intermediate_output` only expects intermediate outputs, not model outputs which could be of type `PIL.Image`.

### Followup From
#2385

jakesabathia2 commented:
Thanks for the PR @Zerui18 ,
could you also include an unittest for this fix?

yixingli-apple commented:
Some references for activation quantization unit tests (you could also put your unit test there):
- https://github.com/apple/coremltools/blob/32ebc557d692e6e7d28d5dc799f21c2afdcd654e/coremltools/test/optimize/coreml/test_passes.py#L3774
- https://github.com/apple/coremltools/blob/32ebc557d692e6e7d28d5dc799f21c2afdcd654e/coremltools/test/optimize/coreml/test_passes.py#L3565


References for getting a mlmodel with ImageType as output:
- https://github.com/apple/coremltools/blob/32ebc557d692e6e7d28d5dc799f21c2afdcd654e/docs-guides/source/image-inputs.md?plain=1#L135
- https://github.com/apple/coremltools/blob/32ebc557d692e6e7d28d5dc799f21c2afdcd654e/coremltools/converters/mil/frontend/torch/test/test_torch_conversion_api.py#L1923

Zerui18 commented:
@jakesabathia2  I've added a corresponding unit test and rebased both changes to be a single commit on top of the current head. Please help to review it, thank you.

jakesabathia2 reviewed (approved):
LGTM!
CI: https://gitlab.com/coremltools1/coremltools/-/pipelines/1593429673

jakesabathia2 commented:
@Zerui18 the CI is failing.
I am seeing this branch is based on an old commit:
```
commit 3d7bfc8281fcfca31ad5a06f7203f763af2a5462
Author: M_noAria <46508354+M-Quadra@users.noreply.github.com>
Date:   Fri Nov 1 01:39:36 2024 +0800

    Enhance torch.zeros_like for GPU inference in dynamic shape (#2369)
```
Can you rebase on the latest `origin/main` ?

Zerui18 commented:
Hi @jakesabathia2, I have rebase my commit on the current head as requested! Please have a look again. Thanks!

jakesabathia2 reviewed (dismissed):
https://gitlab.com/coremltools1/coremltools/-/pipelines/1630212840

TobyRoseman reviewed (changes requested):
CI failures need to be addressed.

jakesabathia2 commented:
Another CI if I rebase on the tot main:
https://gitlab.com/coremltools1/coremltools/-/pipelines/1630494025

jakesabathia2 commented:
The CI is green based on the tot commit,
need t o wait https://github.com/apple/coremltools/pull/2433 to be merged first.

Zerui18 commented:
@jakesabathia2 no problem I have rebased on the latest head, please help to review it.

jakesabathia2 commented:
Thank you @Zerui18 
The test is failing after rebase:
https://gitlab.com/coremltools1/coremltools/-/jobs/8926921705
Prob a utils you were calling got renamed or removed ^
Can you help fix it :)

jakesabathia2 commented:
https://gitlab.com/coremltools1/coremltools/-/pipelines/1640419174

Zerui18 commented:
It looks like there was a problem during the environment setup for the executorch test, but I'm not sure how to fix that since this commit only alters two python scripts.

TobyRoseman commented:
> It looks like there was a problem during the environment setup

This transient issue happens occasionally. I've restarted that job.

jakesabathia2 reviewed (approved):
CI passed
